csi/controller.go

```diff
@@ -1107,6 +1133,41 @@ func (s *OsdCsiServer) DeleteSnapshot(
 	return &csi.DeleteSnapshotResponse{}, nil
 }
 
+func (s *OsdCsiServer) deleteCloudBackup(
+	ctx context.Context,
+	req *csi.DeleteSnapshotRequest,
+) (*csi.DeleteSnapshotResponse, error) {
+	cloudBackupClient, err := s.getCloudBackupClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get any labels passed in by the CO
+	_, locator, _, err := s.specHandler.SpecFromOpts(req.GetSecrets())
```

Comment:
Can we pass locator that we already got on line 1088? Instead of fetching a new copy